File: Lógica de Programación/#04 CADENAS DE CARACTERES/Python/Ejercicio1.py
'''
1️⃣ Contador de Vocales y Consonantes
Consigna:
Crea un programa en Python que solicite al usuario una palabra o frase y cuente:
✅ Cantidad de vocales (a, e, i, o, u).
✅ Cantidad de consonantes.
✅ Cantidad de espacios en blanco (si la entrada es una frase).
'''

def palabra():
    palabra = str(input("Ingrese una cadena de texto: ")).lower()
    contVocal = 0
    contConsonante = 0
    saltos = 0
    for caracter in palabra:
        if caracter in "aeiou":
            contVocal = contVocal + 1
        elif caracter.isalpha():
            contConsonante = contConsonante + 1
        elif caracter.isspace():
            saltos = saltos + 1
    
    print(f"Cantidad de vocales: {contVocal}")
    print(f"Cantidad de consonantes: {contConsonante}")
    print(f"Cantidad de espacios en blanco: {saltos}")


palabra()

File: Lógica de Programación/#04 CADENAS DE CARACTERES/C/Ejercicio1.c
/*Ejercicio 1: Contar vocales
Escribe un programa que pida una cadena al usuario y muestre cuántas vocales (a, e, i, o, u) contiene.*/
#include <stdio.h>
#include <string.h>
#define MAX_CAPACIDAD 100

int main() {
    char cadena[MAX_CAPACIDAD];
    int contador = 0;
    printf("Ingrese una cadena: ");
    if (fgets(cadena, MAX_CAPACIDAD, stdin) == NULL) return 1;
    cadena[strcspn(cadena, "\n")] = '\0';

    for (int i = 0; cadena[i] != '\0'; i++) {
        if (cadena[i] == 'A' || cadena[i] == 'E' || cadena[i] == 'I' || cadena[i] == 'O' || cadena[i] == 'U' || cadena[i] == 'a' || cadena[i] == 'e' || cadena[i] == 'i' || cadena[i] == 'o' || cadena[i] == 'u') {
            contador++;
        }
    }

    printf("\n\nLa cantidad de vocales es: %d", contador);
    return 0;
}